src/features/hostapp/hooks/target-hostapp.ts

```diff
@@ -55,7 +58,46 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 			);
 		}
 	},
-});
+};
+
+const postHostappHook: hooks.Hooks = {
+	async POSTPARSE(args) {
+		if (
+			args.request.values.os_version != null &&
+			args.request.values.os_variant != null
+		) {
+			let deviceType = args.request.values.is_of__device_type;
+			if (!deviceType) {
+				[{ id: deviceType }] = await args.api.get({
+					resource: 'device_type',
+					options: {
+						$select: 'id',
+						$filter: {
+							slug: args.request.values.device_type,
+						},
+					},
+				});
+			}
+			if (!deviceType) {
+				return;
+			}
```

Comment:
Oh I just realized I made a mistake here, this'll break if there's no match
```suggestion
				const [ dt ] = await args.api.get({
					resource: 'device_type',
					options: {
						$select: 'id',
						$filter: {
							slug: args.request.values.device_type,
						},
					},
				});
			}
			if (!dt) {
				return;
			}
			deviceType = dt.id
```